### PR TITLE
Initial structure

### PR DESCRIPTION
**Pull request by sarahsturgeon**

### CODE REVIEW

File: lua/autorun/client/cl_cfc_notifications.lua [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    local MessageLabel = vgui.Create( "DLabel", Bubble )
+    MessageLabel:SetPos( 5, 5 )
+    MessageLabel:SetSize( bubbleX * 0.9, bubbleY * 0.9 )
+    MessageLabel:SetWrap( true )
+    MessageLabel:SetFont( "Trebuchet24" )
+    MessageLabel:SetText( message )
+
+    function MessageLabel:PerformLayout()
+        Title:SetFGColor( Color( 255, 255, 255, 255 ) )
```
Comment (justinmuskopf) [inline lua/autorun/client/cl_cfc_notifications.lua:26]: Brings up an interesting point.
We should add dev-specific values (e.g. `TRANSPARENT = Color(0, 0, 0, 0, 0)`) to `CFCColorLib`.

File: lua/autorun/server/sv_cfc_notifications.lua [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    local getPlayers = filter or player.GetAll
+
+    local players = getPlayers()
+
+    net.Start( "CFC_PopupNotification" )
+        net.WriteString( message )
+    net.Send( players )
+end
+
+CFCNotifications._registerNotification = function( notificationType, message, delay, recurring, filter )
```
Comment (justinmuskopf) [inline lua/autorun/server/sv_cfc_notifications.lua:25]: Is the `_` us politely asking people not to mess with it, or does Lua have some form of encapsulation?

File: lua/autorun/server/sv_cfc_notifications.lua [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    local repititions = recurring and 0 or 1
+
+    local callback = nil
+    if notificationType == "popup" then
+        callback = makePopup
+    end
```
Comment (justinmuskopf) [inline lua/autorun/server/sv_cfc_notifications.lua]:
```suggestion
if notificationType == "popup" then callback = makePopup end
```

File: lua/autorun/server/sv_cfc_notifications.lua [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    net.Start( "CFC_PopupNotification" )
+        net.WriteString( message )
+    net.Send( players )
+end
+
+CFCNotifications._registerNotification = function( notificationType, message, delay, recurring, filter )
+    local repititions = recurring and 0 or 1
+
+    local callback = nil
+    if notificationType == "popup" then
```
Comment (justinmuskopf) [inline lua/autorun/server/sv_cfc_notifications.lua:29]:
```suggestion
    if notificationType == "popup" then callback = makePopup end
```

File: lua/autorun/server/sv_cfc_notifications.lua [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        callback = makePopup
```
Comment (justinmuskopf) [inline lua/autorun/server/sv_cfc_notifications.lua:30]:
```suggestion
```

File: lua/autorun/server/sv_cfc_notifications.lua (hunk 3 above)
Comment (justinmuskopf) [inline lua/autorun/server/sv_cfc_notifications.lua:31]:
```suggestion
```

File: lua/autorun/server/sv_cfc_notifications.lua (hunk 5 above)
Comment (justinmuskopf) [reply lua/autorun/server/sv_cfc_notifications.lua:30]: This is gross.
Can't find a way to make a multiline suggestion...

File: lua/autorun/server/sv_cfc_notifications.lua [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    return function()
+        return { ply }
+    end
+end
+
+local function addTestNotif( ply )
+    CFCNotifications.registerPopup( "This is a test notification!", 1, generateFilter( ply ))
+end
+
+concommand.Add( "cfc_notif", addTestNotif )
```
Comment (justinmuskopf) [inline lua/autorun/server/sv_cfc_notifications.lua:57]: `cfc_notif`? _Really?_

```suggestion
concommand.Add( "cfc_test_notification", addTestNotif )
```

**Review by justinmuskopf — CHANGES_REQUESTED** [review]
Some styling and suggested concommand change

File: lua/autorun/server/sv_cfc_notifications.lua (hunk 2 above)
Comment (sarahsturgeon) [reply lua/autorun/server/sv_cfc_notifications.lua:25]: Just seems like good practice to underscore methods that no one else would intend to use

File: lua/autorun/server/sv_cfc_notifications.lua (hunk 4 above)
Comment (brandonsturgeon) [reply lua/autorun/server/sv_cfc_notifications.lua]: can you make this a multiline comment so I can commit it

File: lua/autorun/server/sv_cfc_notifications.lua (hunk 6 above)
Comment (sarahsturgeon) [reply lua/autorun/server/sv_cfc_notifications.lua:57]: This was for development, it should actually be removed

File: lua/autorun/server/sv_cfc_notifications.lua (hunk 3 above)
Comment (brandonsturgeon) [reply lua/autorun/server/sv_cfc_notifications.lua]: can you make these a

File: lua/autorun/server/sv_cfc_notifications.lua (hunk 5 above)
Comment (sarahsturgeon) [reply lua/autorun/server/sv_cfc_notifications.lua:30]: click and drag on the +'s next to the line numbers

**Pull request merged**